portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,90 @@
 
 package com.google.sps.servlets;
 
+import java.util.*;
 import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter; 
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
+public class DataServlet extends HttpServlet 
+{
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException 
+    {
+        Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService savedComments = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = savedComments.prepare(query);
+
+        List<Task> tasks = new ArrayList<>();
+
+        for (Entity entity : results.asIterable()) 
+        {
+            long id = entity.getKey().getId();
+            long timestamp = (long) entity.getProperty("timestamp");
+            String comment = (String) entity.getProperty("comment");
+            
+            Task task = new Task(id, timestamp, comment);
+            tasks.add(task);
+        }
+
+        Gson gson = new Gson();
+
+        response.setContentType("application/json;");
+        response.getWriter().println(gson.toJson(tasks));
+    }
+
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException 
```

Comment:
It's good practice to add documentation describing your public methods, classes, interfaces etc. For non-public entities, user your judgement: if the entity is doing something interesting / nuanced / non-obvious, then it's probably a good idea to document. Makes it a lot easier for other software engineers to collaborate with you.
This applies to other files in this pull request, and to your future code.